processing/src/main/java/io/druid/query/metadata/SegmentMetadataQueryRunnerFactory.java

```diff
@@ -146,7 +147,9 @@ public SegmentMetadataQueryRunnerFactory(
                           @Override
                           public Sequence<SegmentAnalysis> call() throws Exception
                           {
-                            return input.run(query, responseContext);
```

Comment:
@nishantmonu51 ChainedExecutionQueryRunner seems to be doing same. see here
https://github.com/druid-io/druid/blob/master/processing/src/main/java/io/druid/query/ChainedExecutionQueryRunner.java#L130

pls let me know if you're suggesting something different?